tools/src/tensil/tools/compiler/MemoryDimensions.scala

```diff
@@ -167,33 +167,90 @@ class MemoryDimensions private (
       dimension
   }
 
-  def buildConsts(build: (Option[Int]) => Unit): Unit = {
+  def buildConsts(
+      sourceShape: Seq[Int],
+      broadcast: Boolean,
+      groupSize: Option[Int],
+      build: (Option[Int]) => Unit
+  ): Unit = {
+    require(sourceShape.size == order)
+
     def atLayout(i: Int) =
       atVectors(layout(i)) * (if (i == order - 1) arraySize else 1)
 
     def modelPos(layoutPos: Int*) =
       (0 until layoutPos.size).map(i => layoutPos(layout.indexOf(i)))
 
+    /**
```

Comment:
Just curious: if/when we do a more optimized version of depth-wise convolutions, will this `group size` logic still be necessary?

Reply:
It won't be necessary since we will be doing dense MatMul.